### PR TITLE
fabtests/multinode: update arguments

### LINKED ISSUE #9330: fabtests/multinode: Multinode command line parameters are not documented
fi_multinode has test specific parameters which are not documented in either the test output (e.g. using -? option) or the man pages.  To make it worse, the test re-uses some common command line parameters, so the commont output from -? is wrong.

### PR DESCRIPTION
**Pull request by alex-mckinley**
Update arguments to avoid conflicts with other tests.

Will fix #9330

### CODE REVIEW

**Comment by alex-mckinley** [comment]
```
Usage:
    fi_multinode -s SERVER_NAME -n NUM_RANKS [OPTIONS]
    repeat command for each client
    reccomend using fabtests/scripts/runmultinode.sh

fi_multinode specific options: 

-n <num_ranks>                 Number of ranks to expect
-x <xfer_mode>                 msg or rma message mode
-I <iters>                     number of iterations
-T                             pass to enable performance timing mode
-z <pattern>                   full_mesh, ring, gather, or broadcast pattern. Default: All

General Fabtests options: 

-f <fabric>                    fabric name
-d <domain>                    domain name
-p <provider>                  specific provider name eg sockets, verbs
-B <src_port>                  non default source port number
-P <dst_port>                  non default destination port number
-s <address>                   source address
-b[=<oob_port>]                enable out-of-band address exchange and synchronization over the, optional, port
-E[=<oob_port>]                enable out-of-band address exchange only over the, optional, port
-C <number>                    simultaneous connections to server
-O <addr>                      use the provided addr for out of band
-F <addr_format>               Address format (default:FI_FORMAT_UNSPEC)
-D <device_iface>              Specify device interface: e.g. cuda, ze, neuron (default: None). Automatically enables FI_HMEM (-H)
-i <device_id>                 Specify which device to use (default: 0)
-H                             Enable provider FI_HMEM support
-R                             Register HMEM memory with fi_mr_dmabuf API
```


New fi_multinode opts/fi_multinode -h output

**Comment by j-xiong** [comment]
There are a few more places that need to be updated with the new option (`-C` --> `-x`):
`fabtests/pytest/default/test_multinode.py`,
`fabtests/test_configs/tcp/io_uring.exclude`.

File: fabtests/multinode/src/harness.c [hunk 1]
```diff
@@ -321,17 +321,20 @@ int main(int argc, char **argv)
 	if (!hints)
 		return EXIT_FAILURE;
 
-	while ((c = getopt(argc, argv, "n:C:z:Th" CS_OPTS INFO_OPTS)) != -1) {
+	while ((c = getopt(argc, argv, "n:x:z:Ths:" INFO_OPTS)) != -1) {
```
Comment (j-xiong) [inline fabtests/multinode/src/harness.c:324]: Here a new option "-s" is added, but not used later.

Comment (alex-mckinley) [reply fabtests/multinode/src/harness.c:324]: nvm, you are correct that -I isn't in INFO_OPTS. That also needs to be added manually.

**Review by j-xiong — APPROVED** [review]

**Pull request merged**